src/org/opendatakit/briefcase/util/ServerFetcher.java

```diff
@@ -261,8 +264,8 @@ private ExecutorService getFetchExecutorService(Boolean pullInParallel) {
     return Executors.newFixedThreadPool(downloadThreads, new DownloadThreadFactory());
   }
 
-  private boolean downloadAllSubmissionsForForm(File formInstancesDir, DatabaseUtils formDatabase, BriefcaseFormDefinition lfd,
-                                                FormStatus fs, String websafeCursorString) {
+  private Pair<Boolean, String> downloadAllSubmissionsForForm(File formInstancesDir, DatabaseUtils formDatabase, BriefcaseFormDefinition lfd,
+                                                              FormStatus fs, String websafeCursorString) {
```

Comment:
I’m trying to decide if I’m concerned that these cursors are Strings, and not of a less general type.

Reply:
The ServerFetcher belongs to the legacy codebase. The new codebase will have a `Cursor` abstraction that deals with that (see #705)

Reply:
By the way, the cursor is an XML document that the Aggregate API serializes into another XML's tag value, which is super weird. That's why it's a String at that point.

I think the reasoning was to treat the cursor as an opaque string during HTTP interaction.